### PR TITLE
fix: prevent focus gained/lost loop on wayland

### PR DESCRIPTION
**Pull request by folke**
On wayland, when getting the contents of the system clipboard with `wl-paste`, `wl-paste` is launched.
`wl-paste` gets the window focus, so NEovim `FocusLost` is triggered.
This triggers an endless loop of focus lost/gained.

This PR fixes that.

Fixes https://github.com/gbprod/yanky.nvim/issues/37

### CODE REVIEW

File: lua/yanky.lua [hunk 1]
```diff
@@ -252,7 +252,9 @@ function yanky.on_yank()
   local entry = utils.get_register_info(vim.v.event.regname)
   entry.filetype = vim.bo.filetype
 
-  yanky.history.push(entry)
+  vim.schedule(function()
```
Comment (folke) [inline lua/yanky.lua:255]: this shouldn't be here. let me fix that

**Comment by gbprod** [comment]
Thanks for this PR ! I can't confirm it works but it seems ok!

I'm waiting for some feedback and I'll merge later.

**Comment by lalitmeeq** [comment]
It seems like it's working fine. @gbprod and @folke

**Comment by folke** [comment]
I use Hyprland and didn't have the issue there, probably because `wl-paste` doesn't get focus there when starting.

If you would like to trigger the original issue, open Neovim in Kitty, on Gnome (with wayland) using `wl-clipboard`.

That would create an endless loop and makes Neovim unresponsive.

The fix in this PR is to make sure that loop no longer happens.

Should be safe to merge as is. I can't think of any situation where that kind of behavior would be wanted with the other clipboard handlers.

**Comment by gbprod** [comment]
Thanks !

**Pull request merged**